home/tests/test_content_import_export.py

```diff
@@ -1882,3 +1884,128 @@ def test_list_items(self, impexp: ImportExport) -> None:
+            parent=ha_menu,
+            slug="self-help",
+            title="self-help",
+            bodies=[
+                WABody(
+                    "self-help",
+                    [
+                        WABlk(
+                            "*Self-help programs*",
+                            buttons=[PageBtn("Import Export", page=imp_exp)],
+                        )
+                    ],
+                )
+            ],
+        )
+
+        rev = self_help.save_revision()
+        rev.publish()
+
+        orig = impexp.get_page_json()
+        impexp.export_reimport()
+        imported = impexp.get_page_json()
+        assert imported == orig
+
+    def test_export_missing_go_to_button(self, impexp: ImportExport) -> None:
+        """
+        If a page has a button go to page that no longer exists, the missing button
+        is skipped during export.
+        """
+        home_page = HomePage.objects.first()
+        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+        imp_exp = PageBuilder.build_cpi(home_page, "import-export", "Import Export")
+
+        ha_menu = PageBuilder.build_cp(
+            parent=main_menu,
+            slug="ha-menu",
+            title="HealthAlert menu",
+            bodies=[WABody("HealthAlert menu", [WABlk("*Welcome to HealthAlert*")])],
+        )
+        PageBuilder.build_cp(
+            parent=ha_menu,
+            slug="health-info",
+            title="health info",
+            bodies=[
+                WABody(
+                    "health info",
+                    [
+                        WABlk(
+                            "*Health information*",
+                            buttons=[NextBtn("Next message button")],
+                        )
+                    ],
+                )
+            ],
+        )
+        PageBuilder.build_cp(
+            parent=ha_menu,
+            slug="self-help",
+            title="self-help",
+            bodies=[
+                WABody(
+                    "self-help",
+                    [
+                        WABlk(
+                            "*Self-help programs*",
+                            buttons=[PageBtn("Import Export", page=imp_exp)],
+                        )
+                    ],
+                )
+            ],
+        )
+
+        orig = impexp.get_page_json()
+
+        orig_btn = (
+            # list(orig)[3].get("fields").get("whatsapp_body")[0].get("value").get("buttons")
+            list(orig)[3]["fields"]["whatsapp_body"][0]["value"]["buttons"]
+        )
+
+        # delete page referenced by other content page
+        imp_exp.delete()
+
+        impexp.export_reimport()
+        imported = impexp.get_page_json()
+        imported_btn = list(imported)[3]["fields"]["whatsapp_body"][0]["value"][
+            "buttons"
+        ]
+
+        assert imported != orig
+        assert imported_btn != orig_btn
+        assert imported_btn == []
+        assert orig_btn != []
```

Comment:
In general, asserts should assert on what things *are*, not what they *aren't*.

If we expect the original and reimported pages to be different, we should explicitly modify the original page json to match what we expect the reimported version to look like and assert that it does, otherwise we may miss later bugs that cause the export to change in other ways.